voxblox_ros/include/voxblox_ros/ros_params.h

```diff
@@ -0,0 +1,127 @@
+  nh_private.param("truncation_distance", truncation_distance,
+                   truncation_distance);
+  nh_private.param("max_ray_length_m", integrator_config.max_ray_length_m,
+                   integrator_config.max_ray_length_m);
+  nh_private.param("min_ray_length_m", integrator_config.min_ray_length_m,
+                   integrator_config.min_ray_length_m);
+  nh_private.param("max_weight", max_weight, max_weight);
+  nh_private.param("use_const_weight", integrator_config.use_const_weight,
+                   integrator_config.use_const_weight);
+  nh_private.param("allow_clear", integrator_config.allow_clear,
+                   integrator_config.allow_clear);
+  nh_private.param("start_voxel_subsampling_factor",
+                   integrator_config.start_voxel_subsampling_factor,
+                   integrator_config.start_voxel_subsampling_factor);
+  nh_private.param("max_consecutive_ray_collisions",
+                   integrator_config.max_consecutive_ray_collisions,
+                   integrator_config.max_consecutive_ray_collisions);
+  nh_private.param("clear_checks_every_n_frames",
+                   integrator_config.clear_checks_every_n_frames,
+                   integrator_config.clear_checks_every_n_frames);
+  nh_private.param("max_integration_time_s",
+                   integrator_config.max_integration_time_s,
+                   integrator_config.max_integration_time_s);
+  nh_private.param("anti_grazing", integrator_config.enable_anti_grazing,
+                   integrator_config.enable_anti_grazing);
+  integrator_config.default_truncation_distance =
+      static_cast<float>(truncation_distance);
+  integrator_config.max_weight = static_cast<float>(max_weight);
+
+  return integrator_config;
+}
+
+inline EsdfMap::Config getEsdfMapConfigFromRosParam(
+    const ros::NodeHandle& nh_private) {
+  EsdfMap::Config esdf_config;
+
+  // TODO(helenol): in the future allow different ESDF and TSDF voxel sizes...
+  nh_private.param("tsdf_voxel_size", esdf_config.esdf_voxel_size,
+                   esdf_config.esdf_voxel_size);
+
+  // No specialization for ros param for size_t, so have to do this annoying
+  // workaround.
+  int voxels_per_side = esdf_config.esdf_voxels_per_side;
+  nh_private.param("tsdf_voxels_per_side", voxels_per_side, voxels_per_side);
+  if (!isPowerOfTwo(voxels_per_side)) {
+    ROS_ERROR("voxels_per_side must be a power of 2, setting to default value");
+    voxels_per_side = esdf_config.esdf_voxels_per_side;
+  }
+  esdf_config.esdf_voxels_per_side = voxels_per_side;
+
+  return esdf_config;
+}
+
+inline EsdfIntegrator::Config getEsdfIntegratorConfigFromRosParam(
+    const ros::NodeHandle& nh_private) {
+  EsdfIntegrator::Config esdf_integrator_config;
+
+  // TODO(helenol): in the future allow different ESDF and TSDF voxel sizes...
+  // Make sure that this is the same as the truncation distance OR SMALLER!
+  nh_private.param("tsdf_voxel_size", esdf_integrator_config.min_distance_m,
```

Comment:
I know, you are just moving these parameters, but is it correct to set the min distance in this way?

Reply:
If I'm not mistaken, this value is used to determine the fixed voxels, the lowest frontier? With this current setting that's roughly a single voxel layer around the surface. Not sure if this is the best solution, but seems ok. @helenol ?

Reply:
I agree this is weird... The original thinking was probably a hack workaround for what Marius said (it should be about 1 pixel), but like, this is just a weird way to do it. Will look into fixing this.